client/components/site-selector/index.jsx

```diff
@@ -132,7 +132,9 @@ export default React.createClass( {
 		if ( this.state.search ) {
 			sites = this.props.sites.search( this.state.search );
 		} else {
-			sites = this.shouldShowGroups() ? this.props.sites.getVisibleAndNotRecent() : this.props.sites.getVisible();
+			sites = this.shouldShowGroups()
```

Comment:
Per my previous comment, I think by limiting this to a subset of sites, we need to update [the condition below](https://github.com/Automattic/wp-calypso/blob/d2cb3f10a22d22613a2aa0be27b6eb26afddec7b/client/components/site-selector/index.jsx#L168-L170) to only show "No sites found" if there are in fact no sites at all.